ui/ui-frontend/projects/pastis/src/app/profile/profile-preview/profile-information-tab/profile-information-tab/profile-information-tab.component.ts

```diff
@@ -67,40 +66,50 @@ export class ProfileInformationTabComponent {
 
   submited = false;
 
-  archivalProfileUnit: ArchivalProfileUnit;
-
-  profile: Profile;
-
   private _inputProfile: ProfileDescription;
   pending = false;
 
-  previousValue = (): ProfileDescription => {
-    return this._inputProfile;
-  };
+  updateProfile(inputProfile: ProfileDescription): Observable<ProfileDescription> {
+    const profileDescription = inputProfile;
+    this.setFormFields(profileDescription);
+    if (inputProfile.type === 'PA') {
+      return this.profileService.updateProfilePa(profileDescription as Profile).pipe(catchError(() => of(null)));
+    } else {
+      return this.profileService.updateProfilePua(profileDescription as ArchivalProfileUnit).pipe(catchError(() => of(null)));
+    }
+  }
 
-  isInvalid(): boolean {
-    return false;
+  setFormFields(profileDescription: ProfileDescription) {
+    for (const k of Object.keys(this.form.value)) {
+      if (this.form.value[k]) {
+        // @ts-ignore message not assignable to type 'never'
```

Comment:
pk tu as supprimé cette règle?

Reply:
sinon j'ai un message not assignable to type 'never'
j'ai pas trouvé comment l'empecher : c'est pas le type never c'est un string

Reply:
essayes ce truc est ce que ça marche:
 profileDescription  =  { ...profileDescription, [k as keyof ProfileDescription]: this.form.value[k] };

Reply:
alors j'ai plus le message c'est vrai, mais ca rend tout de meme la chose moins clair

Reply:
bon tu peux  faire un resolve pas grand chose

Reply:
ouai ca te gene pas  ? 
c'etait pour m'éviter d'avoir à lister tout les champs du form et de l'objet que j'ai fait ca - mais si ca gene je reviendrais à ca

Reply:
Aaah dans ce cas là je pense tu peux faire directement un merge

{ ...profileDescription, this.form.value}

mais pas sur que ça marche essayes de le tester

Reply:
j'ai changé et testé 👍